mmaction/datasets/pipelines/augmentations.py

```diff
@@ -696,6 +696,184 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class ColorJitter(object):
+    """Randomly distort the brightness, contrast, saturation and hue of images,
+    and add PCA based noise into images.
+
+    Code Reference:
+    https://gluon-cv.mxnet.io/_modules/gluoncv/data/transforms/experimental/image.html
+    https://mxnet.apache.org/api/python/docs/_modules/mxnet/image/image.html#LightingAug
+
+    If specified to apply color space augmentation, it will distort the image
+    color space by changing brightness, contrast and saturation. Then, it will
+    add some random distort to the images in different color channels.
+    Note that the input images should be in original range [0, 255] and in RGB
+    channel sequence.
+
+    Required keys are "imgs", added or modified keys are "imgs", "eig_val",
+    "eig_vec", "alpha_std" and "color_space_aug".
+
+    Args:
+        color_space_aug (bool): Whether to apply color space augmentations. If
+            specified, it will change the brightness, contrast, saturation and
+            hue of images, then add PCA based noise to images. Otherwise, it
+            will directly add PCA based noise to images. Default: False.
+        alpha_std (float): Std in the normal Gaussian distribution of alpha.
+        eig_val (np.ndarray | None): Eigenvalues of [1 x 3] size for RGB
+            channel jitter. If set to None, it will use the default
+            eigenvalues. Default: None.
+        eig_vec (np.ndarray | None): Eigenvectors of [3 x 3] size for RGB
+            channel jitter. If set to None, it will use the default
+            eigenvectors. Default: None.
+    """
+
+    def __init__(self,
+                 color_space_aug=False,
+                 alpha_std=0.1,
+                 eig_val=None,
+                 eig_vec=None):
+        if eig_val is None:
+            # note that the data range should be [0, 255]
+            self.eig_val = np.array([55.46, 4.794, 1.148])
+        else:
+            self.eig_val = eig_val
+
+        if eig_vec is None:
+            self.eig_vec = np.array([[-0.5675, 0.7192, 0.4009],
+                                     [-0.5808, -0.0045, -0.8140],
+                                     [-0.5836, -0.6948, 0.4203]])
+        else:
+            self.eig_vec = eig_vec
+
+        self.alpha_std = alpha_std
+        self.color_space_aug = color_space_aug
+
+    @staticmethod
+    def brightness(img, delta):
+        """Brightness distortion.
+
+        Args:
+            img (np.ndarray): An input image.
+            delta (float): Delta value to distort brightness.
+                It ranges from [-32, 32).
+
+        Returns:
+            np.ndarray: A brightness distorted image.
+        """
+        if np.random.rand() > 0.5:
+            delta = np.array(delta).astype(np.float32)
```

Comment:
since delta is a float, there is no need to convert it to an array.

```python
img = img + np.float32(delta)
```

is equivalent to these two lines.

Reply:
if one want to speed things up, can write it using inplace operations

```python
# pls be sure that img has dtype of float32
img += delta
```

Reply:
The img may not be in float32 dtype, so the inplace operations may cause error.